### PR TITLE
bgpv1: Documentation update to reflect current architecture

### PR DESCRIPTION
**Pull request by harsimran-pabla**
Updating BGP documentation to match refactor done in #25016.

### CODE REVIEW

File: Documentation/network/bgp-control-plane.rst [hunk 1]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
+
+See the source code at ``pkg/bgpv1/manager/reconcile.go`` for a more in
+depth explanation on how each ``ConfigReconcilerFunc`` is called.
+
+BGP Router
+^^^^^^^^^^
+
+``BGP Control Plane`` utilizes ``gobgp`` as underlying routing agent. 
+
+GoBGP client-side implementation is located here ``pkg/bgpv1/gobgp``.
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane.rst]: Nit (as it is not a link):
```suggestion
GoBGP client-side implementation is located in ``pkg/bgpv1/gobgp``. 
```

File: Documentation/network/bgp-control-plane.rst [hunk 2]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
 Agent-Side Architecture
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Controller
-^^^^^^^^^^
+At a high level, the ``Agent-Side Control Plane`` is divided into following 
+sub-modules
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane.rst]: Nit:
```suggestion
sub-modules:
```

File: Documentation/network/bgp-control-plane.rst [hunk 3]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
 Agent-Side Architecture
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Controller
-^^^^^^^^^^
+At a high level, the ``Agent-Side Control Plane`` is divided into following
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
At a high level, the ``Agent-Side Control Plane`` is divided into the following 
```

File: Documentation/network/bgp-control-plane.rst [hunk 4]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
 -  advertise/withdraw the desired BGP routes
 -  enable/disable any BGP server specific features
 -  inform the caller if the policy cannot be applied
 
-The GoBGP implementation is capable of evaluating each ``CiliumBGPVirtualRouter`` in isolation. This
-means when applying a ``CiliumBGPPeeringPolicy`` the GoBGP ``BGPRouterManager`` will attempt to
-create each ``CiliumBGPVirtualRouter``. If a particular ``CiliumBGPVirtualRouter`` fails to
-instantiate the error is logged and the ``BGPRouterManager`` will continue to the next
-``CiliumBGPVirtualRouter``, utilizing the aforementioned logic.
+The ``BGP Manager`` evaluates each ``CiliumBGPVirtualRouter`` in isolation.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]: Why spaces in `BGP Manager` and `BGP Router`? It does make it slightly easier to read, but not longer corresponds to the real object names in the code, does it?

File: Documentation/network/bgp-control-plane.rst [hunk 5]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
+
+It is worth expanding on how the ``BGP Manager`` works internally.  
+``BGP Manager`` views each ``CiliumBGPVirtualRouter`` as a BGP router instance. 
+Each ``CiliumBGPVirtualRouter`` is defined by a local ASN, a router ID and a 
+list of ``CiliumBGPNeighbors`` with whom it will establish peering.
 
-GoBGP BGPRouterManager Architecture
-***********************************
+This is enough for the ``BGP Manager`` to create a ``BGP Router`` instance. 
+``BGP Manager`` groups ``Bgp Router`` instances by their local ASNs.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
``BGP Manager`` groups ``BGP Router`` instances by their local ASNs. 
```

Although see comment above, not sure about these spaces.

File: Documentation/network/bgp-control-plane.rst [hunk 6]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
+
+   A ``CiliumBGPPeeringPolicy`` applying to a node must not have two or more
+   ``CiliumBGPVirtualRouters`` with the same ``localASN`` fields.
 
-The ``gobgp`` ``BGPRouterManager`` employs a set of ``ConfigReconcilerFunc``\ (s) which perform the
-order-dependent reconciliation actions for each ``BgpServer`` it must reconcile. A
-``ConfigReconcilerFunc`` is simply a function with a typed signature.
+The ``BGP Manager`` employs a set of ``ConfigReconcilerFunc``\ (s) performing an 
+order-dependent reconciliation action for each ``BGP Router`` it must reconcile. 
+A ``ConfigReconcilerFunc`` is a  function with a typed signature.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
A ``ConfigReconcilerFunc`` is a function with a typed signature.
```

What does this info bring, by the way? What type is it?

File: Documentation/network/bgp-control-plane.rst [hunk 7]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
-   type ConfigReconcilerFunc func(ctx context.Context, m *BGPRouterManager, sc *ServerWithConfig, newc *v2alpha1api.CiliumBGPVirtualRouter, cstate *agent.ControlPlaneState) error
+   type ConfigReconcilerFunc func(ctx context.Context, params ReconcileParams) error
+
+See the source code at ``pkg/bgpv1/manager/reconcile.go`` for a more in
+depth explanation on how each ``ConfigReconcilerFunc`` is called.
+
+BGP Router
+^^^^^^^^^^
+
+``BGP Control Plane`` utilizes ``gobgp`` as underlying routing agent.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
``BGP Control Plane`` utilizes ``gobgp`` as the underlying routing agent. 
```

“`gobgp`” or “GoBGP”?

File: Documentation/network/bgp-control-plane.rst [hunk 8]
```diff
@@ -337,69 +337,87 @@ a BGP control plane.
...
+See the source code at ``pkg/bgpv1/manager/reconcile.go`` for a more in
+depth explanation on how each ``ConfigReconcilerFunc`` is called.
+
+BGP Router
+^^^^^^^^^^
+
+``BGP Control Plane`` utilizes ``gobgp`` as underlying routing agent. 
+
+GoBGP client-side implementation is located in ``pkg/bgpv1/gobgp``.
+Implementation API adheres to ``Router`` interface defined in ``pkg/bgpv1/types/bgp.go``.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
Implementation API adheres to the ``Router`` interface defined in ``pkg/bgpv1/types/bgp.go``.
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks! Looks good overall, with some comments below.

File: Documentation/network/bgp-control-plane.rst (hunk 4 above)
Comment (harsimran-pabla) [reply Documentation/network/bgp-control-plane.rst]: In the code, it is BGPRouterManager and Router. I can drop BGP prefix if it makes it easier to read. Any thoughts?

File: Documentation/network/bgp-control-plane.rst (hunk 6 above)
Comment (harsimran-pabla) [reply Documentation/network/bgp-control-plane.rst]: Hmm, it was already present in the document. I do agree that it does not bring much info in the documentation. I will drop this entirely.

File: Documentation/network/bgp-control-plane.rst (hunk 4 above)
Comment (qmonnet) [reply Documentation/network/bgp-control-plane.rst]: OK. Or just take the “BGP” part out of the preformatted text in that case, so it's easy to spot the part that corresponds to existing symbols:

```rst
The BGP ``Manager`` ...
```

File: Documentation/network/bgp-control-plane.rst [hunk 9]
```diff
@@ -337,69 +337,83 @@ a BGP control plane.
...
+
+The ``Manager`` employs a set of ``Reconcilers`` which perform an
+order-dependent reconciliation action for each ``Router``.
+
+
+See the source code at ``pkg/bgpv1/manager/reconcile.go`` for a more in
+depth explanation on how each ``Reconcilers`` works.
+
+Router
+^^^^^^^^^^
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
Router
^^^^^^
```

File: Documentation/network/bgp-control-plane.rst (hunk 6 above)
Comment (harsimran-pabla) [reply Documentation/network/bgp-control-plane.rst]: Replaced it with `Reconcilers`, which is a more accurate description of what is going on.

**Review by qmonnet — APPROVED** [review]
Thanks!

File: Documentation/network/bgp-control-plane.rst (hunk 4 above)
Comment (harsimran-pabla) [reply Documentation/network/bgp-control-plane.rst]: I dropped BGP as a prefix.

**Review by rastislavs — APPROVED** [review]
LGTM

**Comment by harsimran-pabla** [comment]
Documentation change only, relevant docs tests passed and reviews completed. Marking ready-to-merge manually.

**Pull request merged**